crates/protect-ffi/src/lib.rs

```diff
@@ -225,11 +230,21 @@ pub async fn new_client(
         zerokms_config_builder
     };
 
+    let defined_keyset_name = client_opts
+        .keyset_name
+        .map(|name| KeysetIdentifier(IdentifiedBy::Name(name.into())));
+
+    let defined_keyset_id = client_opts
+        .keyset_id
+        .map(|id| KeysetIdentifier(IdentifiedBy::Uuid(id.parse().unwrap())));
+
+    let keyset = defined_keyset_id.or(defined_keyset_name).map(|id| id.0);
+
     let zerokms_config = zerokms_config_builder.build_with_client_key()?;
 
     let zerokms = Arc::new(zerokms_config.create_client());
 
-    let cipher = ScopedZeroKMSNoRefresh::init(zerokms.clone(), None).await?;
+    let cipher = ScopedZeroKMSNoRefresh::init(zerokms.clone(), keyset.clone().or(None)).await?;
```

Comment:
You can just pass the `keyset_id` (which is an `IdentifiedBy` here. Avoids the clone and the `or(None)` is redundant.